vllm/_custom_ops.py

```diff
@@ -599,6 +599,36 @@ def cutlass_scaled_sparse_mm(
     return out
 
 
+def cutlass_fp4_gemm(a: torch.Tensor, b: torch.Tensor, input_sf: torch.Tensor,
+                     weight_sf: torch.dtype, global_sf: torch.dtype,
+                     workspace: torch.dtype, workspace_bytes: int,
+                     out_dtype: torch.dtype) -> torch.Tensor:
+    """
+    Gemm when a and b have nvfp4 datatype(currently represented as a byte),
+    along with their respective block scales and a global scaling factor.
+    """
+    assert (b.shape[0] % 16 == 0 and b.shape[1] % 16 == 0)
+    assert (out_dtype is torch.bfloat16 or out_dtype is torch.float16
+            or out_dtype is torch.float32)
+    m = a.shape[0]
+    n = b.shape[1]
+    workspace_bytes = workspace.nbytes
+    out = torch.empty((m, n), dtype=out_dtype, device=a.device)
+    torch.ops._C.cutlass_fp4_gemm(out, a, b, input_sf, weight_sf, global_sf,
+                                  workspace, workspace_bytes)
+    return out
+
+
+def quantize_to_fp4(input: torch.Tensor, input_sf: torch.Tensor,
```

Comment:
- This should be called `scaled_fp4_quant`
- This should be next to `scaled_fp8_quant` below
- Should `output_sf` really be inplace?